R/weight.R

```diff
@@ -62,12 +123,105 @@ setMethod("weightVariables", "VariableCatalog", function (x) {
     ## Return the names/aliases
     if (length(vars)) {
         return(sort(vapply(index(vars), vget(namekey(x)), character(1),
-            USE.NAMES=FALSE)))
+            USE.NAMES = FALSE)))
     } else {
         return(c())
     }
 })
 
+#' @rdname weightVariables
+#' @export
+setMethod("weightVariables<-", "CrunchDataset", function (x, value) {
+    modifyWeightVariables(x, value, type = "append")
+    return(x)
+})
+
+
+#' modifyWeightVariables
+#'
+#' Change which variables can be set as a dataset's weight.
+#'
+#' @param x a CrunchDataset
+#' @param vars Variables to add or remove  this can be a numeric Crunch variable,
+#' list of numeric Crunch variables or a character vector with the aliases of numeric Crunch variables.
+#' @param type a character string determining how the weightVariables
+#' will be modified:
+#' - `"append"` : add `vars` to the current weight variables
+#' - `"remove"` : remove `vars` from the current list of weight variables
+#' - `"replace"`: replace the current weight variables with `vars`
+#' @return a CrunchDataset
+#' @export
+modifyWeightVariables <- function (x, vars, type = "append") {
+    varcat <- allVariables(x)
+    new <- old <- crGET(shojiURL(varcat, "orders", "weights"))
+    if (is.null(vars)) {
+        new$graph <- NULL
+        type <- "remove"
+    } else {
+        if (is.variable(vars) || (length(vars) == 1) & !is.character(vars)) {
+            vars <- list(vars)
+        }
+        if (is.character(vars)) {
+            ds <- loadDataset(datasetReference(x))
```

Comment:
Could you please add some inline comments explaining why you're doing some of the things you're doing here? There are lots of `if`s and I can't even remember from one review to the next why you're doing some of this this way, so I'm definitely not going to remember next time I look at the code after we're done with this PR.